pkg/validation/internal/multiarch.go

```diff
@@ -384,27 +468,128 @@ func (data *multiArchValidator) checkMissingSupportForOtherImages() {
 				}
 			}
 			if !found && archOSList != "error" {
-				listAllSoNotFound = append(listAllSoNotFound, archOSList)
+				listAllOsNotFound = append(listAllOsNotFound, archOSList)
 			}
 		}
-		if len(listAllSoNotFound) > 0 {
-			sort.Strings(listAllSoNotFound)
+		if len(listAllOsNotFound) > 0 {
+			sort.Strings(listAllOsNotFound)
 			data.warns = append(data.warns,
 				fmt.Errorf("check if the image %s should not support %q. "+
-					"Note that this CSV has labels for this SO(s) "+
+					"Note that this CSV has labels for this OS(s) "+
 					"Your manager image %q are providing this support OR the CSV is configured via labels "+
 					"to support it. Then, please verify if this image should not support it",
 					image,
-					listAllSoNotFound,
+					listAllOsNotFound,
 					data.managerImagesString))
 		}
 	}
 }
 
-// verify if 1 or more allOtherImages has support for a SO not defined via the labels
-// (probably the label for this SO is missing )
-func (data *multiArchValidator) checkMissingLabelsForSO() {
-	notFoundSoLabel := []string{}
+// checkNodeAffinity checks if any image is missing nodeAffinity block corresponding to
+// the supports os/arch platforms in the manifest.
+func (data *multiArchValidator) checkNodeAffinity(images map[string][]platform) {
+	for image, platformFromImage := range images {
+
+		// Verify we were able to gather valid platform data for the image
+		imagePlatformDataValid := true
+		for _, imageData := range platformFromImage {
+			// Ignore the case when the Platform.Architecture == "error" since that means
+			// that was not possible to inspect the image
+			if imageData.Architecture == "error" {
+				imagePlatformDataValid = false
+				break
+			}
```

Comment:
Could you explain why it is okay not to log an error here? Is it logged elsewhere when the image fails to be inspected?

Reply:
It is - specifically it is logged at inspection time in line 382.
https://github.com/operator-framework/api/pull/276/files#diff-20ba073b3d34ec5ee84564e23371a9a0653ca0528bd97fe76837efa1289d8c88R382